builtin/logical/ssh/path_roles.go

```diff
@@ -488,13 +488,113 @@ func (b *backend) getRole(s logical.Storage, n string) (*sshRole, error) {
 	return &result, nil
 }
 
+// parseRole converts a sshRole object into its map[string]interface representation,
+// with appropriate values for each keyType. If the keyType is invalid, it will retun
+// an error.
+func (b *backend) parseRole(role *sshRole) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
```

Comment:
Is this intended to ensure that this method doesn't return a nil? If not, the implicit `make` here can be avoided.

Reply:
Yea since we are assigning whole `result` map in the switch statement, we don't have to initialize it right away.